core/arch/arm/plat-vexpress/conf.mk

```diff
@@ -162,3 +162,4 @@ endif
 endif
 
 CFG_PKCS11_TA ?= y
+CFG_PKCS11_TA_CHECK_VALUE_ATTRIBUTE ?= y
```

Comment:
Why isn't this the global default value?

Reply:
Previous implementation of the pkcs11 TA did provide that support. The added attribute provides a very little bit of hint on the secret value. For generic backward compatibility, we've set it default disabled in _mk/config.mk_.
I would rather ask @vesajaaskelainen if it's fine to default enable that feature.

Reply:
Let's split this comment a bit.

As for whether it should be enabled by default or by "some random machine" then I agree that it should be enabled whenever feature is available so there is no need to make a guess whether feature works in this particular hardware or not.

Now about the CKA_CHECK_VALUE -- personally I feel that the whole feature is a bit dubious as you really cannot use it for anything practical but as it is specified in specification so no objections there. The feature itself is optional.

Now as specified in specification it cannot be used to verify uniqueness of secret key or certificate in anyway -- it is a cue that data might be wrapped/unwrapped properly when transferring the key or so. 

Assuming that we have correct implementation I do not see it as a problem to enable the feature for all.

Now about possibility of leaking key material -- I suppose it is OK to have it as option to make it possible to disable in particular implementation if someone wants to make sure there is no extra path of information leakage. One round of AES-ECB for key itself and taking part of the result in theory gives an advice if someone would be brute forcing the key but the problem itself would be still significant -- same goes for SHA1 for other secret keys and only taking part of the checksum. For certificate we do not care as that information is public.

So from my point of view feel free to enable it by default for all platforms.

Reply:
I've created P-R https://github.com/OP-TEE/optee_os/pull/6579 to fix some issues and default enable CHECK_VALUE support.